pkg/subctl/operator/submarinercr/ensure.go

```diff
@@ -61,10 +63,24 @@ func createSubmariner(clientSet submarinerclientset.Interface, namespace string,
 	} else if errors.IsAlreadyExists(err) {
 		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 			// We can’t always handle existing resources, and we want to overwrite them anyway, so delete them
-			err := clientSet.SubmarinerV1alpha1().Submariners(namespace).Delete(submarinerCR.Name, &metav1.DeleteOptions{})
+			propagationPolicy := metav1.DeletePropagationForeground
+			err := clientSet.SubmarinerV1alpha1().Submariners(namespace).Delete(submarinerCR.Name, &metav1.DeleteOptions{
+				PropagationPolicy: &propagationPolicy,
+			})
 			if err != nil {
 				return fmt.Errorf("failed to delete pre-existing cfg %s : %s", submarinerCR.Name, err)
 			}
+			// We need to wait for the deletion to be complete
+			err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
```

Comment:
10 seconds would be enough?, I've seen resources taking longer to delete, but may be resources which have callbacks ... seems short to me anyway

Reply:
If we bump the timeout then I think we should periodically print a message "_Waiting for deletion of previous Submariner resource %q to complete_".